invoice_restrictions/models/account_move.py

```diff
@@ -0,0 +1,22 @@
+##############################################################################
+#
+#    Copyright (C) 2014-2022 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Emanuel Cino
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+from odoo.exceptions import UserError
+
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _name = 'account.move'
+    _inherit = 'account.move'
+
+    def write(self, vals):
+        if self.mapped('payment_order_id').filtered(lambda p: p.state not in ["draft", "open"]):
```

Comment:
Should we filter for specific vals change only? Wouldn't some changes without impact on the payment order still be possible? I feel this global way is too preventive and might raise errors for cases that are not problematic in the end.